### ISSUE TITLE
smelli silently ignores unknown likelihoods

### ISSUE THREAD

**Issue by MJKirk** [smelli/smelli#25]
When initialising a GlobalLikelihood instance, if your include_likelihoods / exclude_likelihoods arguments contain a likelihood name that smelli doesn't know, no error is raised.
This has caught me when I've made a typo, like typing "fast_likelihoods_quarks.yaml" instead of "fast_likelihood_quarks.yaml" and only when I see the fit results do I discover the mistake. (And only then because it makes a significant difference and I knew the result should change - I could easily imagine some scenario where it would never be noticed.)

My suggestion would be to just add a check that the arguments are contained in _fast_likelihoods_yaml or _likelihoods_yaml and raise an error if not. 
I'm happy to code it myself and submit a pull request if you think this is a good idea.

**Comment by DavidMStraub** [comment]
Hi Matthew!
That's surprising, I thought this line
https://github.com/smelli/smelli/blob/25b1d8ac1c37029caf95eaef087abade38424381/smelli/classes.py#L215
should raise a `FileNotFoundError` in that case. Do you understand why it doesn't? (I cannot check right now.) In any case, a unit test for this case is needed.

**Comment by peterstangl** [comment]
Thank you Matthew for reporting this!
@DavidMStraub, at the moment, the likelihoods are loaded by looping over `self._fast_likelihoods_yaml` and `self._likelihoods_yaml` and a given likelihood is included/excluded if it is found in the `include_likelihoods` / `exclude_likelihoods` arguments. Therefore, a typo in one of the names inside these arguments has the same effect as not including this name.

One way to solve the issue would be to perform a check at the beginning of the `GlobalLikelihood._load_likelihoods` method (e.g. using the subset/superset operator).

**Comment by MJKirk** [comment]
Thanks both for the quick replies - Peter beat me to an explanation!

Doing a subset/superset was my idea for a fix - locally I just put in
```
        if exclude_likelihoods is not None:
            unknown_likelihoods = set(exclude_likelihoods).difference(self._fast_likelihoods_yaml + self._likelihoods_yaml)
            if unknown_likelihoods != set():  
                raise ValueError("exclude_likelihoods contains unknown likelihoods: " + str(unknown_likelihoods))
        if include_likelihoods is not None:
            unknown_likelihoods = set(include_likelihoods).difference(self._fast_likelihoods_yaml + self._likelihoods_yaml)
            if unknown_likelihoods != set():
                raise ValueError("include_likelihoods contains unknown likelihoods: " + str(unknown_likelihoods))
```
at the start of `_load_likelihoods` just as Peter suggests.

**Comment by DavidMStraub** [comment]
Looks good to me! To be more concise, I would recommend to use `if unknown_likelihoods:` and use `... - set(...)` instead of `.difference`.

**Comment by peterstangl** [comment]
You can also use `if exclude_likelihoods:` and `if include_likelihoods:`. And you can combine both checks to avoid duplication of code:

```python
for argument_name, argument in [('exclude_likelihoods', exclude_likelihoods), ('include_likelihoods',include_likelihoods)]:
    if argument:
        unknown_likelihoods = set(argument) - set(self._fast_likelihoods_yaml + self._likelihoods_yaml)
        if unknown_likelihoods:  
            raise ValueError("{} contains unknown likelihoods: {}".format(argument_name,unknown_likelihoods))
```

**Comment by MJKirk** [comment]
Clever!
I guess
https://github.com/smelli/smelli/blob/25b1d8ac1c37029caf95eaef087abade38424381/smelli/test_classes.py#L46
would be the place to add a test for this too?
I can do this, and submit a pull request when it's done.

**Comment by peterstangl** [comment]
Fixed by PR #26.
